kore/src/Kore/Step/Axiom/Matcher.hs

```diff
@@ -668,16 +668,22 @@ matchNormalizedAc
     -> Builtin.NormalizedAc normalized (TermLike Concrete) (TermLike variable)
     -> MaybeT (MatcherT variable unifier) ()
 matchNormalizedAc pushValue wrapTermLike normalized1 normalized2
-  | [] <- opaque2
-  , [] <- excessAbstract1
+  | [] <- excessAbstract1
   = do
     Monad.guard (null excessConcrete1)
     case opaque1 of
-        []       -> do
+        [] -> do
+            Monad.guard (null opaque2)
             Monad.guard (null excessConcrete2)
             Monad.guard (null excessAbstract2)
-        [frame1] -> push (Pair frame1 normalized2')
-        _        -> empty
+        [frame1]
+          | null excessAbstract2
+          , null excessConcrete2
+          , [frame2] <- opaque2 ->
```

Comment:
Would it be difficult to generalize this to accept any number of opaque collections in the second argument? I'm not sure how much that will happen in practice, though.

Reply:
This should be handled below (the otherwise case). Having a single opaque thing means that we can unpack it and have a somewhat optimized unification which is nice if the two opaque things are variables (e.g. we get `varMap1=varMap2` instead of `varMap1=wrap-in-normalized-ac(varMap2)`), but that unwrapping does not work with two opaque things.

Reply:
Oh, I understand now. Thanks!